pkg/logentry/stages/drop.go

```diff
@@ -88,55 +88,71 @@ func newDropStage(logger log.Logger, config interface{}) (Stage, error) {
 	}
 
 	return &dropStage{
-		logger: log.With(logger, "component", "stage", "type", "drop"),
-		cfg:    cfg,
+		logger:    log.With(logger, "component", "stage", "type", "drop"),
+		cfg:       cfg,
+		dropCount: getDropCountMetric(registerer),
 	}, nil
 }
 
 // dropStage applies Label matchers to determine if the include stages should be run
 type dropStage struct {
-	logger log.Logger
-	cfg    *DropConfig
+	logger    log.Logger
+	cfg       *DropConfig
+	dropCount *prometheus.CounterVec
 }
 
-// Process implements Stage
-func (m *dropStage) Process(labels model.LabelSet, extracted map[string]interface{}, t *time.Time, entry *string) {
+func (m *dropStage) Run(in chan Entry) chan Entry {
+	out := make(chan Entry)
+	go func() {
+		defer close(out)
+		for e := range in {
+			if !m.shouldDrop(e) {
```

Comment:
No more weird drop label, we just don't send entries down to the outbound channel.